File: app/src/main/java/com/msapp/wirelessms/interfaces/TTS_Interface.java
package com.msapp.wirelessms.interfaces;

        import android.widget.ImageButton;

        import com.huawei.hms.mlsdk.tts.MLTtsEngine;

public interface TTS_Interface {

    interface ttsPresenter {
        void detectLanguage();
        void init();
        void giveText(String txt);
    }

    interface ttsView{
        void useEngine(MLTtsEngine mlTtsEngine);
        ImageButton ttsButton();
        ImageButton nextButton();
        ImageButton prevButton();
        ImageButton repeatButton();
        String sourceText();
    }
}

File: app/src/main/java/com/msapp/wirelessms/interfaces/TTSInterface.kt
package com.msapp.wirelessms.interfaces

import android.widget.ImageButton
import com.huawei.hms.mlsdk.tts.MLTtsEngine

interface TTSInterface {

    interface TPresenter {
        fun detectLanguage()
        fun init()
        fun giveText(txt: String)
        fun setConfigs(lng: String, gender: String)
        //fun giveCurrentLng(): String?
    }

    interface TView {
        fun useEngine(mlTtsEngine: MLTtsEngine?)
        fun ttsButton(): ImageButton?
        fun sourceText(): String?
        fun listen(str: String)
        fun selectSpeaker(lng: String)
        fun setVolume(): Float
        fun setSpeed(): Float
    }
}